src/config.ts

```diff
@@ -47,6 +50,9 @@ export function readConfig(env: string, config?: string): SyntheticsConfig {
   }
   if (typeof options === 'function') {
     options = options(env);
+    if (options instanceof Promise) {
```

Comment:
We need something like this
```
    options != null
    typeof val.then === 'function' &&
    typeof val.catch === 'function'
```

Promises are transpiled in different ways based on the Native promises and ES6 promises. This would be the safest way to check for Promise like function.